android/src/main/java/com/amazonaws/ivs/reactnative/player/AmazonIvsView.kt

```diff
@@ -371,6 +382,20 @@ class AmazonIvsView(private val context: ThemedReactContext) : FrameLayout(conte
   private fun intervalHandler() {
     val reactContext = context as ReactContext
 
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N
```

Comment:
Can we lean on https://developer.android.com/develop/ui/views/picture-in-picture#handling_ui 
```
override fun onPictureInPictureModeChanged(
     isInPictureInPictureMode: Boolean, newConfig: Configuration) {
```
instead of relying on polling the state?

Reply:
I believe I attempted this but couldn't get it triggering so I went with polling as a last ditch effort as I needed something working at the time.

Reply:
However once the other PiP PR had been merged, I am happy to test it again :)

Reply:
Hey @Bowlerr was hoping to merge this PiP relate PR before https://github.com/aws/amazon-ivs-react-native-player/pull/156 👍

Reply:
Hey @dawhitla,

Last time I attempted to use this, it never triggered / claimed to override nothing.

Is there something I am missing?

Thank you

Reply:
Yeah I can pull your fork and take a look at why `onPictureInPictureModeChanged` does not trigger 👍

Reply:
Have cleaned this up a little replacing:

`if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.N
      && reactContext.packageManager
        .hasSystemFeature(
          PackageManager.FEATURE_PICTURE_IN_PICTURE))`
          
with

`if(pipEnabled)` in the `intervalHandler`

Reply:
Ahh I figured out why `onPictureInPictureModeChanged` isn't happening for the view

It's a function of a callback on the activity itself. `src/main/java/com/facebook/react/ReactActivity.java` 

So hooking into `onPictureInPictureModeChanged` would require a manual setup 😮‍💨 

Here is what another RN lib did
https://github.com/adkaushik/react-native-pip-android#setup

We may need to rely on the interval check for the time being for easier DX (no manual setup step)

Reply:
Ah thank you ! @dawhitla 

I believe that's where I got to before switching to the interval check. Although it isn't ideal, the interval check has been reliable in our production use case :) Check out **[Moar](https://www.moar.vip/getmoar)** if you want to see it in action.

I am happy to look into getting this done at some point but other priorities have taken my focus for the time being. Feel free to merge as is